### DISCUSSION TITLE
how safe is vlang compared with rust? can it be used in mission critical space rocket launching program or even in space?

### DISCUSSION THREAD

**Discussion by sprappcom** [vlang/v#20727] [Questions and Answers] [answered]
how safe is vlang compared with rust? can it be used in mission critical space rocket launching program or even in space?

**Comment by JalonSolov** [comment, marked as answer]
I wouldn't, yet.  V is still only at Beta 0.4.4, so there are still problems to be resolved.

**Comment by Wajinn** [comment]
Looking at this question/these questions, think it should be noted that it can create confusion, as it is 2 separate questions on top of each other.

> How safe is vlang compared with rust? 

Vlang is a much younger and newer language than Rust.  It's not usually going to be a fair comparison, when one language is so much older than the other or was a corporate controlled creation versus a truly individual non-corporate open source effort at first. In addition, they have different purposes behind their creation and different goals, though there is some significant overlap.

From what I've read and understand, Vlang was created to be a safer programming language, but not for that to be its primary mission or to override it being easy to use and understand.

> Can it be used in mission critical space rocket launching program or even in space?

Think that is a bit of a loaded question or at least very difficult to answer in a straight forward way. For instance C and C++ are both languages which have been used for mission critical purposes. Vlang can compile to C, as it's the main backend of several being worked on or planned (the others are native V, JavaScript, WASM, Go, and Pascal).

When C or C++ are used for mission critical purposes, they are often adhering to the MISRA C standard. It might be possible or something team V may want to take on in the future, where V has a "MISRA C conformance" flag, and attempts to check and produce C code that meets the MISRA guidelines.  Something along the lines of Cppcheck. As V is already a safer language, it's likely to be significantly in range of the guidelines.

**Reply by kolinfluence** [reply]
very good answer. "MISRA C conformance" flag shld be the way forward.
